app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -76,15 +83,21 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
             }
         })
 
-        if (Settings.getInstance(this@MainActivity).shouldShowPocketOnboarding()) {
-            val onboardingIntents =
-                    Intent(this@MainActivity, PocketOnboardingActivity::class.java)
-            startActivity(onboardingIntents)
-        }
+        if (!intent.hasExtra("TURBO_MODE")) {
```

Comment:
So you want to be able to inject "TURBO_MODE" check through intent? If so, this isn't the best place to do it. If the Activity is never destroyed (i.e. backgrounded), onCreate() would never get called again. Please see `onNewIntent()`

Reply:
I think you're correct, this should use `onNewIntent`: it makes sense to "re-launch" with a different turbo mode setting.  I'll do that.  I'm going to keep it here too, though, to disable the first-run experience in automation.

Reply:
Sounds good